dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -47,9 +49,21 @@
   private final ClassDumper classDumper;
   private final ImmutableMap<Path, SymbolReferenceSet> jarToSymbols;
   private final ClassReferenceGraph classReferenceGraph;
+  private final ImmutableMap<Path, Artifact> pathToArtifact;
 
   public static LinkageChecker create(List<Path> jarPaths, Iterable<Path> entryPoints)
       throws IOException {
+    return create(jarPaths, entryPoints, null);
+  }
+
+  public static LinkageChecker create(
+      ImmutableMap<Path, Artifact> pathToArtifact, Iterable<Path> entryPoints) throws IOException {
```

Comment:
I notice we're using a *lot* of maps for these sorts of things. This suggests we might be missing an abstraction somewhere.